### PR TITLE
Fix release notes about improper freeze of 1.6

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
#1367 do not properly freeze contract key lookup restriction (#1866) in 1.6
Consequently the restriction is still only available in 1.dev

This PR, fix the release notes and drop the comment about this change.

We also introduce some minor cleanup in the version handling in the engine. We hope reducing the chance of such improper freeze.

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala [hunk 1]
```diff
@@ -532,11 +531,11 @@ private[archive] class DecodeV1(minor: LanguageMinorVersion) extends Decode.OfPa
           CPCons(name(cons.getVarHead), name(cons.getVarTail))
 
         case PLF.CaseAlt.SumCase.OPTIONAL_NONE =>
-          assertSince("1", "CaseAlt.OptionalNone")
+          assertSince(optionalVersion, "CaseAlt.OptionalNone")
```
Comment (gerolf-da) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala]: Why have a "named" alias specifically for version `1.1` but not some others? The name `optionalVersion` also reads more like that the version is optional. Do you want to make the move to "true" feature flags later, or stick with language aliases in `Features`?

File: daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala [hunk 2]
```diff
@@ -178,7 +176,7 @@ private[digitalasset] class EncodeV1(val minor: LanguageMinorVersion) {
           builder.setCon(
             PLF.Type.Con.newBuilder().setTycon(tycon).accumulateLeft(args)(_ addArgs _))
         case TBuiltin(bType) =>
-          if (bType == BTArrow && V1.minorVersionOrdering.lteq(minor, "0")) {
+          if (bType == BTArrow && LV.ordering.lt(version, LV.v1_1)) {
```
Comment (gerolf-da) [inline daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala]: This changed from version `1.0` to version `1.1`.

Comment (remyhaemmerle-da) [reply daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala]: I change in the same time `lteq` by `lt`. I think it is more consistent with the `assertSince` we use in lot of other place.

Comment (gerolf-da) [reply daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala]: Right. I didn't notice that part. Whatever makes it more consistent with other checks works for me.

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala]: I created alias for all version.

**Pull request merged**